### DISCUSSION TITLE
Connectors in airflow

### DISCUSSION THREAD

**Discussion by NonlinearNimesh** [apache/airflow#25598] [General]
Hi, i have a doubt on connectors, Does airflow has a concept of connectors, like i want to create a custom connectors for the sole purpose of reusabllity. But i didnt get anything regarding connectors in airflow.

**Comment by alexkruc** [comment]
I'm not sure what you're referring to exactly because it's too broad. Airflow has a concept of [Connections](https://airflow.apache.org/docs/apache-airflow/stable/howto/connection.html) ([quick overview ](https://airflow.apache.org/docs/apache-airflow/stable/concepts/connections.html#connections-hooks)) that you can define and re-use, and also [Variables](https://airflow.apache.org/docs/apache-airflow/stable/howto/variable.html) which are a simple key-value store that you can set and it will be shared by everything in Airflow..

I hope it helps :)

**Reply by NonlinearNimesh** [reply]
Thanks alexkruc for your response, my objective is to built a connector for a specific task (for example: converting all uppercase words to lowercase) and use this connector wherever we need this.

Perdon me, I might sound clueless as i have just started airflow.

**Reply by alexkruc** [reply]
Airflow doesn't have something like this because each task is different and does different tasks.
If you want, you can create your own `task_common_utils.py` file, with relevant functions in it, and import them to your DAG file and to relevant tasks..

For example:
```python
from task_common_utils import uppercase_to_lowercase

def print_lowercase_string():
    lowercase_string = uppercase_to_lowercase("HELLO WORLD")
    print(lowercase_string)

dag= DAG(......)
with dag:
    my_task = PythonOperator(
        task_id='print_lowercase',
        python_callable=print_lowercase_string
    )

```

From this example, I've imported the function as common code, and I can use it inside the `PythonOperator` code.

BTW if you're new, I recommend you go over the [Tutorial](https://airflow.apache.org/docs/apache-airflow/stable/tutorial.html)

**Reply by potiuk** [reply]
Airflow is Python. You do not need to have "connector"  - you just keep your shared code in the DAGs folder, import it and use it in as many places you want. See https://airflow.apache.org/docs/apache-airflow/stable/modules_management.html?highlight=modules+management for all details.
